clang/lib/Interpreter/Interpreter.cpp

```diff
@@ -738,6 +766,15 @@ Expr *Interpreter::SynthesizeExpr(Expr *E) {
   return Result.get();
 }
 
+std::string Interpreter::getAllInput() const {
+  return IncrParser->AllInput;
+}
+
+std::vector<CodeCompletionResult> Interpreter::CodeComplete(llvm::StringRef Input, size_t Col, size_t Line){
```

Comment:
```suggestion
void Interpreter::CodeComplete(llvm::StringRef Input, size_t Col, size_t Line, const std::vector<CodeCompletionResult> &CompletionResults){
```

This way we can get rid of the `CompletionResults` data member.

Reply:
@vgvassilev The problem here is the `CompletionResults` data member mediates the communication between the `ReplCompletionConsumer::ProcessCodeCompleteResults` and the `ReplCompleter`. The completion consumer is set to the compiler instance when an interpreter is created, regardless of what the interpreter is created for.   Therefore, we need to change a little bit the way an interpreter is created for code completion 

I will push a fix.